src/interp/interp-wasi.cc

```diff
@@ -66,8 +76,49 @@ class WasiInstance {
     return Result::Ok;
   }
 
+  Result fd_fdstat_get(const Values& params, Values& results, Trap::Ptr* trap) {
+    int32_t fd = params[0].i32_;
+    if (trace_stream) {
+      trace_stream->Writef("fd_fdstat_get %d\n", fd);
+    }
+    results[0].i32_ = __WASI_ERRNO_NOENT;
+    return Result::Ok;
+  }
+
+  Result fd_read(const Values& params, Values& results, Trap::Ptr* trap) {
+    int32_t fd = params[0].i32_;
+    int32_t iovptr = params[1].i32_;
+    int32_t iovcnt = params[2].i32_;
+    if (trace_stream) {
+      trace_stream->Writef("fd_read %d [%d]\n", fd, iovcnt);
+    }
+    __wasi_iovec_t* wasm_iovs = getMemPtr<__wasi_iovec_t>(iovptr, iovcnt, trap);
```

Comment:
`getMemPtr` could use an out parameter and return `Result`, then we can use `CHECK_RESULT` here rather than duplicating the `if` statement below.

Reply:
Done.  Nice.